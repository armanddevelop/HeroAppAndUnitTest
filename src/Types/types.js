export const types = {
  login: "[aut] Login",
  logout: "[aut] Logout",
};
