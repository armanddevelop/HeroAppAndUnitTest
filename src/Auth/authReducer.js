import { types } from "../Types/types";

// const initialState = {
//   name:"Licha",
//   logged: true
// }

export const authReducer = (state = {}, action) => {
  console.log("Esto vale action ", action);
  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true,
      };
    case types.logout:
      return {
        logged: false,
      };
    default:
      return state;
  }
};
