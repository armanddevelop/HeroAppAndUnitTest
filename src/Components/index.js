export { HeroList } from "./Hero/HeroList";

export { Hero } from "./Hero/Hero";

export { Navbar } from "./UI/NavBar";

export { Login } from "./Login/Login";

export { SearchScreen } from "./Search/SearchScreen";

export { DcScreen } from "./DC/DcScreen";

export { MarvelScreen } from "./Marvel/MarvelScreen";
