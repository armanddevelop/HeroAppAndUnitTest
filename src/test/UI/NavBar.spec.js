import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../Auth/AuthContext";
import { Navbar } from "../../Components";
import { types } from "../../Types/types";

const mockNavigate = jest.fn();
const dispatch = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));
describe("Test to Nabar Component", () => {
  const contextValue = {
    user: {
      name: "licha",
    },
    dispatch,
  };
  const container = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={["/dc"]}>
        <Navbar />
      </MemoryRouter>
    </AuthContext.Provider>
  );
  it("Snapshot to Navbar component", () => {
    expect(container).toMatchSnapshot();
    expect(container.find(".text-info").text().trim()).toBe(
      contextValue.user.name
    );
  });
  it("Should execute handleLogOut", () => {
    const action = {
      type: types.logout,
    };
    container.find("button").prop("onClick")();

    expect(mockNavigate).toHaveBeenCalledWith("/login", { replace: true });
    expect(dispatch).toHaveBeenCalledWith(action);
  });
});
