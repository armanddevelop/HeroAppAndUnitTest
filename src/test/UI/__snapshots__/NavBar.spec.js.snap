// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to Nabar Component Snapshot to Navbar component 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      Array [
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>,
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>,
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>,
      ]
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        licha
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
